### PR TITLE
Fix portability and dependency issue

### PR DESCRIPTION
**Pull request by aasifkhan7**
1. Adds Cython dependency in requirements_common.txt
2. modifies the name of reindex-solr() in docker/ol-docker-start.sh to reindex_solr() for portability purposes.

### CODE REVIEW

**Comment by tabshaikh** [comment]
Hi @aasifkhan7 Can you please elaborate why the Cython dependency is being added and what are we trying to achieve using this pr

**Comment by aasifkhan7** [comment]
Hi @tabshaikh. The aim is this PR is to address the two things that I found that might be a problem. 
1. While running sh docker/ol-docker-start.sh I found that reindex-solr() function threw a BadName error due to the hyphen. So for portability reasons, we can rename that to reindex_solr().
2. Also, I assumed that we need to run python setup.py install at some point in order to install the openlibrary modules. And that results in an ImportError of Cython. Hence added Cython dependency. 
Please let me know if I'm mistaken in any of the thoughts.

**Comment by cclauss** [comment]
Please resolve conflict.

**Comment by aasifkhan7** [comment]
> Please resolve conflict.

Done @cclauss

**Comment by aasifkhan7** [comment]
I also see `reindex-solr` in 2 more places:
```
./Makefile:62:reindex-solr:
./scripts/2018/10/bootstrap.sh:122:    sudo -u $OL_USER make reindex-solr
```

Should I change here - `./scripts/2018/10/bootstrap.sh:122`? 🤔

**Comment by cclauss** [comment]
I never mess with the scripts/20xx files so they are  probably not Py3-ready.  Let’s leave it to @mekarpeles and @cdrini to decide if that file should be changed.

File: docker/ol-docker-start.sh [hunk 1]
```diff
@@ -53,8 +53,8 @@ EOM
 su openlibrary -c "$PY_ENV_VARS && scripts/infobase-server conf/infobase.yml 7000" &
 
 # wait unit postgres is ready, then reindex solr
-export -f reindex-solr
-su openlibrary -c "$PY_ENV_VARS && until pg_isready; do sleep 5; done && reindex-solr localhost $CONFIG" &
+export -f reindex_solr
```
Comment (cdrini) [inline docker/ol-docker-start.sh:56]: All this code has been deleted :)

File: requirements_common.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
+Cython==0.29.20
```
Comment (cdrini) [inline requirements_common.txt:1]: I believe this dependency is only used by the solrbuilder (and installed inside its Docker file specifically, here: https://github.com/internetarchive/openlibrary/blob/master/scripts/solr_builder/Dockerfile.olpython ). To the best of my knowledge, the setup.py file is never actually run :/

**Review by cdrini — COMMENTED** [review]
I'm going to close this one; the docker-start has been largely modified/simplified, and that function is no longer there. And I believe cython is only used by the Dockerfile.olpython in solr_builder; so we shouldn't need it as a general dependency.

**Pull request closed without merging**